integration-cli/docker_api_containers_test.go

```diff
@@ -1616,3 +1616,10 @@ func (s *DockerSuite) TestPostContainersCreateWithOomScoreAdjInvalidRange(c *che
 		c.Fatalf("Expected output to contain %q, got %q", expected, string(b))
 	}
 }
+
+// test case for #22210 where an emtpy container name caused panic.
+func (s *DockerSuite) TestContainerApiDeleteWithEmptyName(c *check.C) {
+	status, _, err := sockRequest("DELETE", "/containers/", nil)
+	c.Assert(err, checker.IsNil)
+	c.Assert(status, checker.Equals, http.StatusNotFound)
```

Comment:
Thanks @cpuguy83 I just updated the pull request. In addition to the check in `GetContainer(prefixOrName string)` I still placed a check in `GetByName()` in case later on this function is called in other places.